src/libraries/System.Security.Cryptography.Xml/src/System/Security/Cryptography/Xml/CanonicalXmlElement.cs

```diff
@@ -12,8 +12,12 @@ internal sealed class CanonicalXmlElement : XmlElement, ICanonicalizableNode
     {
         private bool _isInNodeSet;
 
-        public CanonicalXmlElement(string prefix, string localName, string namespaceURI, XmlDocument doc, bool defaultNodeSetInclusionState)
+        public CanonicalXmlElement(string? prefix, string localName, string? namespaceURI, XmlDocument doc, bool defaultNodeSetInclusionState)
+#if NET7_0_OR_GREATER
             : base(prefix, localName, namespaceURI, doc)
+#else
+            : base(prefix!, localName, namespaceURI, doc)
+#endif
```

Comment:
honestly, probably better to always add `!` than have conditional compile here. It will result in the same IL in the end

Reply:
Yes,  we shouldn't use conditionals just to avoid a `! `. `! ` has zero impact on the IL generated.